manticore/utils/emulate.py

```diff
@@ -99,88 +119,140 @@ def __init__(self, cpu):
         self.registers = set(self._cpu.canonical_registers)
         # The last 8 canonical registers of x86 are individual flags; replace with the eflags
         self.registers -= self.flag_registers
-        # TODO(eric_k): unicorn@1.0.2rc1 doesn't like writing to
-        # the FS register, and it will segfault or hang.
-        self.registers -= {"FS"}
         self.registers.add("EFLAGS")
 
-        for reg in self.registers:
-            val = self._cpu.read_register(reg)
-
-            if reg in {"FS", "GS"}:
-                self.msr_write(reg, val)
-                continue
-
-            if issymbolic(val):
-                from ..native.cpu.abstractcpu import ConcretizeRegister
-
-                raise ConcretizeRegister(
-                    self._cpu, reg, "Concretizing for emulation.", policy="ONE"
-                )
-            logger.debug("Writing %s into %s", val, reg)
-            self._emu.reg_write(self._to_unicorn_id(reg), val)
-
-        for m in cpu.memory.maps:
-            self.map_memory_callback(m.start, len(m), m.perms, m.name, 0, m.start)
+        self.load_state_from_manticore()
 
     def reset(self):
         self._emu = Uc(self._uc_arch, self._uc_mode)
         self._to_raise = None
 
-    def copy_memory(self, address, size):
+    def copy_memory(self, address: int, size: int):
         """
         Copy the bytes from address to address+size into Unicorn
         Used primarily for copying memory maps
         :param address: start of buffer to copy
         :param size: How many bytes to copy
         """
         start_time = time.time()
-        map_bytes = self._cpu._raw_read(address, size)
+        map_bytes = self._cpu._raw_read(address, size, force=True)
```

Comment:
Was the lack of `force=True` causing a soundness problem? It seems like it shouldn't prevent `rusticorn` from executing correctly because the old behavior would have been to raise `InvalidMemoryAccess`. Unless this is an older change meant to support one of the CHESS programs?

Reply:
> Was the lack of force=True causing a soundness problem?

Yes. I was seeing the program change some memory maps to zero permissions, which would cause Manticore to throw `InvalidMemoryAccess` when reading from those maps. Since this is an internal implementation detail to synchronize our emulator memory, I think `force=True` is warranted here.